fda/grid.py

```diff
@@ -366,7 +399,8 @@ def derivative(self, order=1):
         dataset_label = "{} - {} derivative".format(self.dataset_label, order)
 
         return FDataGrid(data_matrix, sample_points, self.sample_range,
```

Comment:
With so many places where we have to copy the object and change the matrix, maybe it pays off to just implement `__copy__` to share whatever objects are immutable in a shallow copy, and use `copy.copy()` instead of manually constructing the object with the same arguments in each place. I think it would be less error-prone.

Reply:
It's a good idea to consider in a future pull request.